rfc/0006-threat-model.md

```diff
@@ -0,0 +1,262 @@
+
+## Threat ID 13 - Attacker exploits bad string matching on a blocklist to  bypass rules
+An attacker, who has rights to create workloads, bypasses a rule by exploiting
+bad string  matching.
+
+**Mitigation**
+All rules are reviewed and tested.
+
+**To do**
+We may came up with some tests to cover this rules reviews
+We  should carefully review a PR changing the rules in the policies deployment
+
+## Threat ID 14 - Attacker uses new/old features of the Kubernetes API  which have no rules
+An attacker, with rights to create workloads, uses new features of the Kubernetes
+API (for  example a changed API version) to bypass a rule.
+
+**Mitigation**
+All rules are reviewed and tested.
+
+**To do**
+We may came up with some tests to cover this rules reviews and API versions.
+We can create a configuration to reject by default requests where the API
+version not cover by the policy.  We should warning policies developers to cover
+all the supported API version in theirs tests and reject all of others.
+
+
+## Threat ID 15 - Attacker deploys privileged container to node running  Webhook controller
+An attacker, who has rights to deploy privileged containers to the cluster, creates
+a privileged  container on the cluster node where the admission controller webhook operates.
+
+**Mitigation**
+Admission controller uses restrictive policies to prevent privileged  workloads
+
+**To do**
+We have a policy to prevent privileged workloads in the recommended policies in
+the `kubewarden-default` helm chart. But we can improve by:
+- We may need to define all the API versions cover by the recommended policies
+- We can create a configuration to reject by default requests where the API version not cover by the policy.
+- We should warning policies developers to cover all the supported API version in theirs tests and reject all of others.
+
+
+## Threat ID 16 - Attacker mounts a privileged node hostpath allowing  modification of Webhook controller configuration
+An attacker, who has rights to deploy hostPath volumes with workloads, creates a
+volume  which allows for access to the admission controller pod’s files.
+
+**Mitigation**
+Admission controller uses restrictive policies to prevent privileged  workloads
+
+**To do**
+We can add a recommended policy in the `kubewarden-default` Helm chart to prevent this.
+
+
+## Threat ID 17 - Attacker has privileged SSH access to cluster node  running admission webhook
+An attacker is able to log into cluster nodes as a privileged user via SSH.
+**Mitigation**
+N/A
+
+I don't think Kubewarden can help on this
+
+
+## Threat ID 18 - Attacker uses policies to send confidential data from  admission requests to external systems
+An attacker is able to configure a policy that listens to admission requests and
+sends  sensitive data to an external system.
+
+**Mitigation**
+Strictly control external access for webhook
+
+**To do**
+We may define some network configuration to block external access from the Kubewarden pods.
```

Comment:
Policies run in a very restrictive environment, I think while we should have this threat in consideration it does not apply that much to Kubewarden.

Reply:
Agree, this doesn't apply to Kubewarden policies because we don't offer network access to our Wasm policies.

Reply:
:+1: